client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -169,14 +169,14 @@
                     .add("row", io.trino.client.Row.class, Row.class, (type, clientRow) -> (Row) convertFromClientRepresentation(type, clientRow))
                     .add("row", io.trino.client.Row.class, Map.class, (type, clientRow) -> {
                         Row row = (Row) convertFromClientRepresentation(type, clientRow);
-                        Map<String, Object> result = new HashMap<>();
-                        for (RowField field : row.getFields()) {
-                            String name = field.getName()
-                                    .orElseGet(() -> "field" + field.getOrdinal());
-                            if (result.containsKey(name)) {
+                        List<RowField> fields = row.getFields();
+                        Map<String, Object> result = new HashMap<>(fields.size() + (fields.size() / 2));
+                        for (RowField field : fields) {
+                            Optional<String> fieldName = field.getName();
+                            String name = fieldName.isPresent() ? fieldName.get() : "field" + field.getOrdinal();
```

Comment:
It's a capturing lambda that captures the loop variable, so an extra allocation on each iteration that could be avoided. I changed this slightly to in the latest version because of a bug uncovered by the unit tests.